tsl/test/sql/update_exclusion.sql

```diff
@@ -0,0 +1,167 @@
+DECLARE
+  tbl name;
+BEGIN
+  FOR tbl IN SELECT 'metrics_' || unnest(ARRAY['int2','int4','int8','date','timestamp','timestamptz','space'])
+  LOOP
+    EXECUTE format('ALTER TABLE %I DROP COLUMN %I;', tbl, $1);
+  END LOOP;
+END;
+$$;
+
+-- create 4 chunks each with different physical layout
+SELECT drop_column('c1');
+INSERT INTO metrics_int2(time) VALUES (0);
+INSERT INTO metrics_int4(time) VALUES (0);
+INSERT INTO metrics_int8(time) VALUES (0);
+INSERT INTO metrics_date(time) VALUES ('2000-01-01');
+INSERT INTO metrics_timestamp(time) VALUES ('2000-01-01');
+INSERT INTO metrics_timestamptz(time) VALUES ('2000-01-01');
+INSERT INTO metrics_space(time,device) VALUES ('2000-01-01','1'),('2000-01-01','2');
+
+SELECT drop_column('c2');
+INSERT INTO metrics_int2(time) VALUES (10);
+INSERT INTO metrics_int4(time) VALUES (10);
+INSERT INTO metrics_int8(time) VALUES (10);
+INSERT INTO metrics_date(time) VALUES ('2001-01-01');
+INSERT INTO metrics_timestamp(time) VALUES ('2001-01-01');
+INSERT INTO metrics_timestamptz(time) VALUES ('2001-01-01');
+INSERT INTO metrics_space(time,device) VALUES ('2001-01-01','1'),('2001-01-01','2');
+
+SELECT drop_column('c3');
+INSERT INTO metrics_int2(time) VALUES (20);
+INSERT INTO metrics_int4(time) VALUES (20);
+INSERT INTO metrics_int8(time) VALUES (20);
+INSERT INTO metrics_date(time) VALUES ('2002-01-01');
+INSERT INTO metrics_timestamp(time) VALUES ('2002-01-01');
+INSERT INTO metrics_timestamptz(time) VALUES ('2002-01-01');
+INSERT INTO metrics_space(time,device) VALUES ('2002-01-01','1'),('2002-01-01','2');
+
+SELECT drop_column('c4');
+INSERT INTO metrics_int2(time) VALUES (30);
+INSERT INTO metrics_int4(time) VALUES (30);
+INSERT INTO metrics_int8(time) VALUES (30);
+INSERT INTO metrics_date(time) VALUES ('2003-01-01');
+INSERT INTO metrics_timestamp(time) VALUES ('2003-01-01');
+INSERT INTO metrics_timestamptz(time) VALUES ('2003-01-01');
+INSERT INTO metrics_space(time,device) VALUES ('2003-01-01','1'),('2003-01-01','2');
+
+SELECT drop_column('c5');
+
+-- immutable constraints
+-- should not have ChunkAppend since constraint is immutable and postgres already does the exclusion
+-- should only hit 1 chunk and base table
+BEGIN;
+:PREFIX UPDATE metrics_int2 SET value = 0.5 WHERE time = 15;
+:PREFIX UPDATE metrics_int4 SET value = 0.5 WHERE time = 15;
+:PREFIX UPDATE metrics_int8 SET value = 0.5 WHERE time = 15;
+ROLLBACK;
+
+-- stable constraints
+-- should have ChunkAppend since constraint is stable
+-- should only hit 1 chunk and base table
+BEGIN;
+:PREFIX UPDATE metrics_int2 SET value = 0.3 WHERE time = length(substring(version(),1,23));
+:PREFIX UPDATE metrics_int4 SET value = 0.3 WHERE time = length(substring(version(),1,23));
+:PREFIX UPDATE metrics_int8 SET value = 0.3 WHERE time = length(substring(version(),1,23));
+ROLLBACK;
+
+-- should have ChunkAppend since constraint is stable
+-- should only hit 1 chunk and base table, toplevel rows should be 1
+BEGIN;
```

Comment:
To be exhaustive, I would also add a test with multiple column updates 
UPDATE <hypertable> SET col1 =xxx,  col2 = yyy, col3 = zzz
WHERE ....
RETURNING ...

Reply:
Will it work also with writable CTEs?

Reply:
added a couple more tests